radix-substate-store-impls/src/state_tree/tree_store.rs

```diff
@@ -177,7 +170,7 @@ pub trait WriteableTreeStore {
 /// avoid this potential performance implication by having an explicit way of associating an
 /// "unchanged" Substate with its new tree leaf.
 pub enum AssociatedSubstateValue<'v> {
-    Upserted(&'v DbSubstateValue),
+    Upserted(&'v [u8]),
```

Comment:
I quite like the well-named types instead of arbitrary bytes. Even if it's just an alias, it documents things a bit.

(minor; you can say "but here it literally means arbitrary bytes" and I will be ok)

Reply:
Yeah broadly I agree with you! :)

However, here `DbSubstateValue = Vec<u8>` - I wanted to change it to the more natural `&[u8]` rather than using `&Vec<u8>`.

I don't really like the type alias pattern for these sorts of types because they don't protect anything; I prefer new types.

Arguably that's what `AssociatedSubstateValue` is already :) - however, imo really it should wrap a `RawScryptoValue<'a>` - that's essentially what this PR tries to do: https://github.com/radixdlt/radixdlt-scrypto/pull/1860